### PR TITLE
Fix docs qt6 wasm mobile

### LINKED ISSUE #637: Documentation and logs say I need to install desktop Qt alongside Android Qt5
**Describe the document error or request**
In several places, the documentation and log messages state that it is necessary to install the desktop version of Qt alongside the Android version of Qt, regardless of whether I am using Qt5 or Qt6. I have reason to believe that this is only true for Qt6 and not Qt5.

I have tested a few versions of android Qt (not exhaustively), and I have noticed that android Qt 5.15.2 can build projects without a parallel desktop installation, while every version I have tried above Qt6 cannot. The `Updater.patch_qmake` code already contains logic that supports this idea: it uses the parallel desktop Qt installation for Qt6 and ignores it for Qt5.

**Proposed change**
Rewrite the documentation and log messages to state which versions of Qt for Android require a parallel desktop version of Qt.

**Additional context**
I am pretty sure that I wrote this documentation and the log messages, so I only have myself to blame! 🤣

### PR DESCRIPTION
**Pull request by ddalcino**
Fix #637.

~Please note that this pulls in two commits from #634, otherwise the tests will not pass.~ **Edit:** rebased onto master

### CODE REVIEW

File: aqt/installer.py [hunk 1]
```diff
@@ -1055,9 +1055,8 @@ def _get_autodesktop_dir_and_arch(
         self, should_autoinstall: bool, host: str, target: str, base_path: Path, version: Version, is_wasm: bool = False
     ) -> Tuple[Optional[str], Optional[str]]:
         """Returns expected_desktop_arch_dir, desktop_arch_to_install"""
-        is_wasm_qt6 = target == "desktop" and is_wasm and version >= Version("6.0.0")
-        if target not in ["ios", "android"] and not is_wasm_qt6:
-            # We do not need to worry about the desktop directory if target is not mobile, or it's not Qt6 wasm.
+        if version < Version("6.0.0") or (target not in ["ios", "android"] and not is_wasm):
```
[the remark this answers is not in the document]
Comment (ddalcino) [reply aqt/installer.py:1058]: @sonatype-lift ignore

**Comment by ddalcino** [comment]
There's a new error in CI for wasm builds; looks like there's a new version of emsdk that doesn't like the fastcomp library we're using:

> ./emsdk install sdk-fastcomp-1.38.27-64bit
error: the fastcomp backend is no longer supported.  Please use an older version of emsdk (for example 3.1.29) if you want to install the old fastcomp-based SDK

**Comment by ddalcino** [comment]
Two builds failed due to timeouts while installing Qt and tools_mingw

**Comment by miurahr** [comment]
Thank you.

**Pull request merged**